bot/exts/utilities/colour.py

```diff
@@ -0,0 +1,262 @@
+import colorsys
+import json
+import pathlib
+import random
+import string
+from io import BytesIO
+from typing import Optional
+
+import discord
+import rapidfuzz
+from PIL import Image, ImageColor
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+THUMBNAIL_SIZE = (80, 80)
+
+
+class Colour(commands.Cog):
+    """Cog for the Colour command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with open(pathlib.Path("bot/resources/utilities/ryanzec_colours.json")) as f:
+            self.colour_mapping = json.load(f)
+            del self.colour_mapping['_']  # Delete source credit entry
+
+    async def send_colour_response(self, ctx: commands.Context, rgb: tuple[int, int, int]) -> None:
+        """Create and send embed from user given colour information."""
+        name = self._rgb_to_name(rgb)
+
+        try:
+            colour_or_color = ctx.invoked_parents[0]
+        except IndexError:
+            colour_or_color = "colour"
+
+        colour_mode = ctx.invoked_with
+        if colour_mode == "random":
+            colour_mode = colour_or_color
+        elif colour_mode in ("colour", "color"):
+            input_colour = rgb
+        elif colour_mode == "name":
+            input_colour = ctx.kwargs["user_colour_name"]
+        elif colour_mode == "hex":
+            input_colour = ctx.args[2:][0]
+            if len(input_colour) > 7:
+                input_colour = input_colour[0:-2]
+        else:
+            input_colour = tuple(ctx.args[2:])
+
+        if colour_mode not in ("name", "hex", "random", "color", "colour"):
+            colour_mode = colour_mode.upper()
+        else:
+            colour_mode = colour_mode.title()
+
+        colour_embed = discord.Embed(
+            title=colour_or_color.title(),
+            description=f"{colour_mode} information for `{name or input_colour}`.",
```

Comment:
I don't think this description reflects what information is actually provided in the embed.

> Name information for Black
> Hex information for Red

In all cases above where the command succeeds, all six values for the color is shown (RGB, HEX, name, HSL, etc). Whereas the description implies only the name or hex values (in the examples above) is in the embed.

How about:

> Colo(u)r information for HEX `#71653e`
> Colo(u)r information for RGB `(1, 2, 3)`
> Colo(u)r information for color `White`

Where:
* `Colo(u)r` is taken from `colour_or_color.title()` like before.
* The last value can still be `name or input_colour`, but if users run `.color #556677` (and a matching name isn't found) that value should then be `#556677` which can be taken from `ctx.kwargs["colour_input"]` (or `color_input` if you decided not to update that).

And perhaps we could set the title to the user input value (or the color name if matched any), like the last value in the examples above?

Hopefully this screenshot summarizes what I've said here (forgot to switch to dark mode, sorry):

![D8954720-CCA0-4432-8B17-F2B040030902](https://user-images.githubusercontent.com/50042066/141604732-dcd706b4-a2d3-48d7-b365-b809dfe93133.jpeg)

Reply:
Fixed in: e87e037 and e9cda06

Now the embeds look like your examples above:

![Screenshot from 2021-11-13 10-19-15](https://user-images.githubusercontent.com/42116429/141649218-5f90db58-8caf-4738-967c-7c010a417a22.png)

![Screenshot from 2021-11-13 10-19-51](https://user-images.githubusercontent.com/42116429/141649251-3ae73f60-9d06-4a6a-9f57-3d418b21d2e7.png)

![Screenshot from 2021-11-13 10-19-58](https://user-images.githubusercontent.com/42116429/141649252-a84c8f97-c35f-4806-a346-2954b36e45cc.png)

![Screenshot from 2021-11-13 10-20-09](https://user-images.githubusercontent.com/42116429/141649254-d37a56fd-18c9-41a7-bf9a-b6020c32f2ff.png)
![Screenshot from 2021-11-13 10-20-29](https://user-images.githubusercontent.com/42116429/141649257-e2b75818-eed9-422e-b293-342a2009f1c4.png)